src/Simulation.jl

```diff
@@ -538,4 +539,57 @@ function rates(bsol::Q;ts::X=Array{Float64,1}()) where {Q<:Simulation,X<:Abstrac
     return hcat([rates(bsol,t) for t in ts]...)
 end
 
+"""
+calculate the rates of all reactions at time t
+"""
+function rates(ssys::Q,t::X) where {Q<:SystemSimulation,X<:Real}
+    rts = zeros(length(ssys.reactions))
+    domains = getfield.(ssys.sims,:domain)
+    Nrxns = sum([length(sim.domain.phase.reactions) for sim in ssys.sims])+sum([length(inter.reactions) for inter in ssys.interfaces if hasproperty(inter,:reactions)])
+    Nspcs = sum([length(sim.domain.phase.species) for sim in ssys.sims])
+    cstot = zeros(Nspcs)
+    vns = Array{Any,1}(undef,length(domains))
+    vcs = Array{Any,1}(undef,length(domains))
+    vT = Array{Any,1}(undef,length(domains))
+    vP = Array{Any,1}(undef,length(domains))
+    vV = Array{Any,1}(undef,length(domains))
+    vC = Array{Any,1}(undef,length(domains))
+    vN = Array{Any,1}(undef,length(domains))
+    vmu = Array{Any,1}(undef,length(domains))
+    vkfs = Array{Any,1}(undef,length(domains))
+    vkrevs = Array{Any,1}(undef,length(domains))
+    vHs = Array{Any,1}(undef,length(domains))
+    vUs = Array{Any,1}(undef,length(domains))
+    vGs = Array{Any,1}(undef,length(domains))
+    vdiffs = Array{Any,1}(undef,length(domains))
+    vCvave = Array{Any,1}(undef,length(domains))
+    vphi = Array{Any,1}(undef,length(domains))
+    index = 1
+    for (k,sim) in enumerate(ssys.sims)
+        vns[k],vcs[k],vT[k],vP[k],vV[k],vC[k],vN[k],vmu[k],vkfs[k],vkrevs[k],vHs[k],vUs[k],vGs[k],vdiffs[k],vCvave[k],vphi[k] = calcthermo(sim.domain,ssys.sol(t),t)
+        cstot[sim.domain.indexes[1]:sim.domain.indexes[2]] = vcs[k]
+        rts[index:index+length(vkfs[k])-1] .= getrates(sim.domain.rxnarray,cstot,vkfs[k],vkrevs[k]).*getdomainsize(sim,t)
+        index += length(vkfs[k])
+    end
+    for inter in ssys.interfaces
+        if hasproperty(inter,:reactions)
+            kfs,krevs=getkfskrevs(inter,vT[inter.domaininds[1]],vT[inter.domaininds[2]],vphi[inter.domaininds[1]],vphi[inter.domaininds[2]],vGs[inter.domaininds[1]],vGs[inter.domaininds[2]],cstot)
+            rts[index:index+length(kfs)-1] = getrates(inter.rxnarray,cstot,kfs,krevs).*inter.A
```

Comment:
So @views is useful if you have subarrays involved in a calculation you want to avoid making copies of, here we are only assigning values to rts rather than using it in a calculation so even without views it won't make a copy of rts.